mlrun/api/api/endpoints/workflows.py

```diff
@@ -0,0 +1,423 @@
+            workflow_name=workflow_spec.name,
+            arguments=workflow_spec.args,
+            source=project.spec.source,
+            kind=load_and_run_fn.kind,
+        )
+
+        if workflow_spec.schedule:
+            # This logic follows the one is performed in `BaseRuntime._enrich_run()`
+            meta_uid = uuid.uuid4().hex
+
+            # creating runspec for scheduling:
+            run_object_kwargs = {
+                "spec": {
+                    "scrape_metrics": config.scrape_metrics,
+                    "output_path": (artifact_path or config.artifact_path).replace(
+                        "{{run.uid}}", meta_uid
+                    ),
+                },
+                "metadata": {"uid": meta_uid, "project": project.metadata.name},
+            }
+
+            runspec = _create_run_object_for_workflow_runner(
+                project=project,
+                workflow_spec=workflow_spec,
+                artifact_path=artifact_path,
+                namespace=namespace,
+                **run_object_kwargs,
+            )
+
+            # Creating scheduled object:
+            scheduled_object = {
+                "task": runspec.to_dict(),
+                "schedule": workflow_spec.schedule,
+            }
+
+            # Creating schedule:
+            get_scheduler().create_schedule(
+                db_session=db_session,
+                auth_info=auth_info,
+                project=project.metadata.name,
+                name=load_and_run_fn.metadata.name,
+                kind=mlrun.api.schemas.ScheduleKinds.job,
+                scheduled_object=scheduled_object,
+                cron_trigger=workflow_spec.schedule,
+                labels=load_and_run_fn.metadata.labels,
+            )
+
+            return {
+                "project": project.metadata.name,
+                "name": workflow_spec.name,
+                "status": "scheduled",
+                "schedule": workflow_spec.schedule,
+            }
+
+        else:
+            runspec = _create_run_object_for_workflow_runner(
+                project=project,
+                workflow_spec=workflow_spec,
+                artifact_path=artifact_path,
+                namespace=namespace,
+                workflow_name=run_name,
+                workflow_handler=workflow_spec.handler,
+            )
+
+            run = load_and_run_fn.run(
+                runspec=runspec,
+                local=False,
+                schedule=workflow_spec.schedule,
+                artifact_path=artifact_path,
+            )
+
+            state = mlrun.run.RunStatuses.running
+
+            return {
+                "project": project.metadata.name,
+                "name": workflow_spec.name,
+                "status": state,
+                "run_id": run.uid(),
+            }
+
+    except Exception as err:
+        logger.error(traceback.format_exc())
+        log_and_raise(HTTPStatus.BAD_REQUEST.value, reason=f"runtime error: {err}")
+
```

Comment:
Is this really necessary? this is being done by default ( logs the error and returns `BAD_REQUEST`)

Reply:
I would do it if you had different cases response cases you wanted to implement